clippy_lints/src/inconsistent_struct_constructor.rs

```diff
@@ -66,6 +66,7 @@ declare_lint_pass!(InconsistentStructConstructor => [INCONSISTENT_STRUCT_CONSTRU
 impl LateLintPass<'_> for InconsistentStructConstructor {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'_>) {
         if_chain! {
+            if !expr.span.from_expansion();
```

Comment:
You could use `clippy_utils::in_macro` for this purpose, though this wouldn't affect the result in the case.